tests/integration/test_charm.py

```diff
@@ -120,3 +120,26 @@ async def test_get_primary_action(ops_test: OpsTest) -> None:
 
         # assert get-primary returned the right primary
         assert identified_primary == expected_primary
+
+
+async def test_exactly_one_primary_reported_by_juju(ops_test: OpsTest) -> None:
+    primary = []
```

Comment:
since this variable is defined outside the scope of the function `check`, it will increase to a length of 2 when a new primary gets elected and hence the call to `check()` on line 142 will fail.

Perhaps instead you can put `primary` within the scope of `check` and then return the variable `primary`